File: apps/admin_mobile/lib/infrastructure/api/mappers/client_resource_mapper.dart
import '../../../domain/entities/client_resource.dart';

ClientResource mapClientResource(Map<String, dynamic> json) {
  return ClientResource(
    id: json['id'] as int,
    tenantId: json['tenant_id'] as int,
    clientId: json['client_id'] as int,
    data: json['data'] as Map<String, dynamic>?,
    plate: json['plate'] as String?,
    brand: json['brand'] as String?,
    model: json['model'] as String?,
    color: json['color'] as String?,
    type: json['type'] as String?,
    createdAt: DateTime.parse(json['created_at'] as String),
    clientName:
        json['client']?['name'] as String? ?? json['client_name'] as String?,
    clientEmail:
        json['client']?['email'] as String? ?? json['client_email'] as String?,
  );
}

File: apps/admin-v2/src/infrastructure/api/mappers/client-resource.mapper.ts
import type { ClientResource } from '@/domain/entities/client-resource';

export function mapClientResource(raw: Record<string, unknown>): ClientResource {
  const client = raw.client as Record<string, unknown> | undefined;

  return {
    id: raw.id as string,
    tenantId: raw.tenant_id as string,
    clientId: raw.client_id as string,
    label: (raw.label as string) ?? null,
    data: (raw.data as Record<string, unknown>) ?? null,
    plate: (raw.plate as string) ?? null,
    brand: (raw.brand as string) ?? null,
    model: (raw.model as string) ?? null,
    color: (raw.color as string) ?? null,
    type: (raw.type as string) ?? null,
    createdAt: new Date(raw.created_at as string),
    client: client
      ? {
          name: client.name as string,
          email: client.email as string,
        }
      : undefined,
  };
}
